### ISSUE TITLE
Ability to restore Skype language

### ISSUE THREAD

**Issue by arhohryakov** [dlehn/Skype-Utility-Project#11]
At present Skype doesn't restore language that has been chosen from its menu. Can this feature be implemented?

**Comment by nestdimon** [comment]
SUP 1.3.3: works fine.
Setting stored at c:\Users\%YOUR_PROFILE%\AppData\Roaming\Skype\sup.ini

**Comment by arhohryakov** [comment]
SUP menu language works fine. Agree. But Skype itself can't restore its language.
On my PC local language is Russian. I choose in Skype menu -> Tools -> Language -> English.
But after restart language is still Russian.
I'm using SkypePortable. There is a note:

> Skype uses an embedded Internet Explorer to show ads and other information that will leave cache files on the local PC. Skype Portable will use the language setting of the local PC on each launch and not save changes due to a bug in Skype itself.

Can it be handled by SUP?

**Comment by dlehn** [comment]
Why did you close this, is it no longer an issue?  
In theory, it should be possible to implement this feature.

**Comment by arhohryakov** [comment]
I've chosen wrong button :[ (in reference issue closing)

Please, implement it.

With gratitude.

**Comment by dlehn** [comment]
Alright, will do. But it will probably take until the weekend, I've got a bachelor's thesis to finish first :unamused:

**Comment by arhohryakov** [comment]
Thanks in advance.

**Comment by dlehn** [comment]
Sorry I'm a little late here, I've had a lot of stuff on my mind the last few days. Just wanted to let you know that I didn't forget about this problem, I started working on it today.  
The whole thing is a little more difficult than I anticipated, though, thanks to Skype's lazy menu initialization. Anyway, I'm sure I'll be able to figure something out.

**Comment by arhohryakov** [comment]
Don't worry. I'll wait.

**Comment by arhohryakov** [comment]
Is there any progress?

**Comment by dlehn** [comment]
I don't have much to show yet, I'm afraid. 
Saving Skype's language setting is easy and works flawlessly, but I can't figure out how to restore it after restarting Skype. The problem is that Skype will ignore the language change command, unless the language popup menu is currently being displayed.
Meaning that if I manually show that menu and then send the matching command, I should in theory be able to force Skype's language to change. However, for some reason or another, I just can't seem to get it to work. No matter what I try, Skype will always either
- Crash
- Exit without comment
- Or simply open the menu but ignore the change language command I send

I have no f*cking idea what I'm doing wrong. Why will Skype accept the command if the menu was opened manually by the user, but not when I open the menu from my code?  

TL;DR: I'm stumped.

**Comment by arhohryakov** [comment]
Thank you for your insistency anyway. 8)
